src/main/java/com/hunllefhelper/HunllefHelperPlugin.java

```diff
@@ -99,12 +93,13 @@ protected void shutDown() throws Exception
 	@Subscribe
 	public void onGameTick(GameTick tick)
 	{
-		if (!config.autoHide())
+		if (config.visibilityMode() == VisibilityMode.AlwaysVisible)
 		{
 			return;
 		}
 
-		boolean isInInstance = isInTheGauntlet();
+		boolean isInInstance = config.visibilityMode() == VisibilityMode.OnlyInHunllefRoom
+				? isInHunllefRoom() : isInTheGauntlet();
 		if (isInInstance != wasInInstance)
```

Comment:
I think these variables should be renamed. The `InInstace` part is logical when auto hiding based on being in the gauntlet instances. Perhaps `wasPanelVisible` and `shouldPanelBeVisible` or something similair are better names.

I'm also fine with leaving the names as they are. I think I will refactor the code a bit later to move part of the hiding logic to a separate component.